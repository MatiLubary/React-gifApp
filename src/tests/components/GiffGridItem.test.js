import React from 'react'
import {GifGridItem} from '../../components/GifGridItem';
import { shallow } from 'enzyme';


describe('Pruebas en <GifGridItem />', () => {

    const title = "un titulo";
    const url = "https://localhost/algo.jpg"
    const wrapper = shallow( <GifGridItem title={title} url={url}/>);
    

    test('should display <GifGridItem /> correctly', () => {
            

        expect(wrapper).toMatchSnapshot();
    });


    test('should have a paragrapgh with a title', () => {
            
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);


    });

    test('should have img equal to url and alt of props', () => {
            
        const img = wrapper.find("img");
        console.log(img.props().src)

        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);

    });

    test('should havethe animate fade in class', () => {
            
        const div = wrapper.find("div");
        console.log(div.props())

        
        expect(div.hasClass("animate__backInLeft")).toBe(true);

    });


})