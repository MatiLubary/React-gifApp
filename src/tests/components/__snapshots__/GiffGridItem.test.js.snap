// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should display <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
